src/MPMCQueue.NET/MPMCQueue.cs

```diff
@@ -38,18 +38,17 @@ public bool TryEnqueue(object item)
         {
             do
             {
-                var buffer = _buffer;
```

Comment:
I would keep this read, because it reads the first field in the class which is CPU cache friendlier.
```
 -00007ffe`162b0d40 57              push    rdi
 -00007ffe`162b0d41 56              push    rsi
 -00007ffe`162b0d42 55              push    rbp
 -00007ffe`162b0d43 53              push    rbx
 -00007ffe`162b0d44 4883ec28        sub     rsp,28h
 -00007ffe`162b0d48 488b7108        mov     rsi,qword ptr [rcx+8]
 -00007ffe`162b0d4c 8b7948          mov     edi,dword ptr [rcx+48h]
 -00007ffe`162b0d4f 8b4110          mov     eax,dword ptr [rcx+10h]
```

vs 

```
 +00007ffb`3a790690 57              push    rdi
 +00007ffb`3a790691 56              push    rsi
 +00007ffb`3a790692 53              push    rbx
 +00007ffb`3a790693 4883ec20        sub     rsp,20h
 +00007ffb`3a790697 8b7148          mov     esi,dword ptr [rcx+48h]
 +00007ffb`3a79069a 8bfe            mov     edi,esi
 +00007ffb`3a79069c 237910          and     edi,dword ptr [rcx+10h]
 +00007ffb`3a79069f 488b4108        mov     rax,qword ptr [rcx+8]
```

Reply:
yes, you are right, didn't know it has this benefit.